### PR TITLE
Simplifying the internal data model of Completion.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
1. 'PresentationItems' removed entirely.  This existed purely so we could add 'snippet' information to a completion item description.  That's something we can do in a vastly simpler manner.
2. Simplified how we deal with 'suggestion items' in the completion model.
3. Removed redundant data that could be computed from other data.

### CODE REVIEW

File: src/EditorFeatures/Core/Implementation/IntelliSense/Completion/DescriptionModifyingPresentationItem.cs [hunk 1]
```diff
@@ -1,73 +0,0 @@
...
-            var description = await this.CompletionService.GetDescriptionAsync(document, this.Item, cancellationToken).ConfigureAwait(false);
-            var parts = description.TaggedParts;
-
-            var change = await GetTextChangeAsync(this.CompletionService, document, this.Item, '\t').ConfigureAwait(false);
-            var insertionText = change.NewText;
-
-            var note = string.Empty;
-            if (snippetService != null && snippetService.SnippetShortcutExists_NonBlocking(insertionText))
-            {
-                note = string.Format(FeaturesResources.Note_colon_Tab_twice_to_insert_the_0_snippet, insertionText);
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Implementation/IntelliSense/Completion/DescriptionModifyingPresentationItem.cs]: This code moved into the C#/VB completion service, where it belonged.

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by rchande** [comment]
👍

**Comment by Pilchie** [comment]
@mattwar - do you want to take a look?

File: src/Features/Core/Portable/Completion/Providers/AbstractMemberInsertingCompletionProvider.cs [hunk 2]
```diff
@@ -253,9 +253,7 @@ internal virtual CompletionItemRules GetRules()
             return s_defaultRules;
         }
 
-        public override Task<CompletionDescription> GetDescriptionAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
-        {
-            return MemberInsertionCompletionItem.GetDescriptionAsync(item, document, cancellationToken);
-        }
+        protected override Task<CompletionDescription> GetDescriptionWorkerAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
+            => MemberInsertionCompletionItem.GetDescriptionAsync(item, document, cancellationToken);
```
Comment (mattwar) [inline src/Features/Core/Portable/Completion/Providers/AbstractMemberInsertingCompletionProvider.cs]: Use normal method form.

**Comment by mattwar** [comment]
:shipit:

File: src/Features/Core/Portable/Completion/Providers/AbstractMemberInsertingCompletionProvider.cs (hunk 2 above)
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/Completion/Providers/AbstractMemberInsertingCompletionProvider.cs]: :D

**Pull request merged**